ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -279,14 +317,12 @@ def take_step(self, env: BaseUnityEnvironment, curr_info: AllBrainInfo):
 
         # If any lessons were incremented or the environment is
         # ready to be reset
-        if self.meta_curriculum and any(lessons_incremented.values()):
-            curr_info = self._reset_env(env)
-            for brain_name, trainer in self.trainers.items():
-                trainer.end_episode()
-            for brain_name, changed in lessons_incremented.items():
-                if changed:
-                    self.trainers[brain_name].reward_buffer.clear()
-
+        if ( ((self.meta_curriculum)  and any(lessons_incremented.values()))
```

Comment:
This is way too complicated. If you're actually checking what the comment says, then split it up into something like
```
lessons_were_incremented = ...
ready_for_reset = ...
if lessons_were_incremented or ready_for_reset:
    # do stuff
```
(also watch the modulo by zero)

Reply:
Added a check to ensure that global_step isn't 0 to ensure modulo safety